RESTserver/src/main/java/com/softserve/if072/restservice/service/ForecastService.java

```diff
@@ -0,0 +1,316 @@
+    private final ProductService productService;
+    private static final long MILLISECONDS_PER_DAY = 24 * 60 * 60 * 1000;
+
+    public ForecastService(HistoryDAO historyDAO, ProductDAO productDAO,
+                           StorageDAO storageDAO, ProductService productService) {
+        this.historyDAO = historyDAO;
+        this.productDAO = productDAO;
+        this.storageDAO = storageDAO;
+        this.productService = productService;
+    }
+    /*
+    * General forecasting strategy:
+    * We can forecast the date of the product ending having average speed of the product using per day (or other time span)
+    * and the product amount in the storage. We can also following the next steps for calculating the average speed of the
+    * product using per day :
+    * 1. Retrieve the data (date and amount) about the product using from the history table.
+    * Since the size of the data won't be changed and we'll be working with primitives (int, double),
+    * store the data in the arrays like "Timestamp[] usedDates" for the dates and
+    * "int[] usedAmounts" for the amounts is the best choice.
+    * 2. Calculate the time spans in milliseconds between two nearby dates in such way:
+    * timeSpans[i-1]=usedDates[i].getTime()-usedDates[i-1].getTime(); (i=0, 1,..., usedDates.length). In this case,
+    * a timeSpans array will be one less than a usedDates array and we should correctly map the time spans to the
+    * appropriate used amounts (timeSpans[0] will be mapped to usedAmounts[1] and so on).
+    * 3. Since the time spans between two nearby dates can be differ through the dates set we should transform them in to
+    * the one time span like day : daysDiff[i]=timeSpans[i]/MILLISECONDS_PER_DAY.
+    * 4. Calculate the speeds of the product using for each time span in the daysDiff array.
+    * 5. Calculate the main descriptive statistics values for the productUsedSpeeds array : mean, min, max, count, range,
+    * standard deviation, lower and upper three sigma limits.
+    * 6. If standard deviation is not equals to 0, then the product used speeds set could contain outliers in the data and
+    * we should detect them and deal with them. According to the theory of the normal data distribution, 99.73% of the values
+    * lie within three standard deviations of the mean or within the lower and upper three sigma limits. Therefore, we can
+    * convince that the data values that are beyond three standard deviations from the mean (less than the lower three sigma
+    * limit or more than the upper three sigma limit) are outliers with the probability of 0.9973.
+    * 7. Detect the data that are less than the lower three sigma limit or more than the upper three sigma. If there are the
+    * outliers in the data set we can deal with them in two ways:
+    *  - if the outliers occur at the beginning of the data set or at the end of the data set we can simply remove them (trim
+    * the data set);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)